lua/amp/lockfile.lua

```diff
@@ -1,5 +1,59 @@
 local M = {}
 
+-- Simple path join that handles platform separators
+local function joinpath(...)
+	local sep = package.config:sub(1, 1)
+	local parts = { ... }
+	-- Trim any trailing/leading separators to avoid double separators
+	for i, p in ipairs(parts) do
+		-- Normalize separators to OS-specific sep
+		p = tostring(p)
+		p = p:gsub("[/\\]", sep)
+		if i > 1 then
+			p = p:gsub("^" .. sep .. "+", "")
+		end
+		if i < #parts then
+			p = p:gsub(sep .. "+$", "")
+		end
+		parts[i] = p
+	end
+	return table.concat(parts, sep)
+end
+
+local function homedir()
+	return vim.loop.os_homedir() or vim.fn.expand("~")
```

Comment:
Just wondering, practically, when will `vim.loop.os_homedir` not have a value?  And if it does, probably a lot of other stuff are going to break as well?

Reply:
According to Amp:

  • Missing HOME (Unix) or USERPROFILE (Windows) environment variables                                   
  • Minimal containers or chroot environments without proper user setup                                  
  • Permission issues accessing user directory information                                               
  • Corrupted system user database

I agree, there's probably something fundamental broken if we reach that point.